vault/datadog_checks/vault/vault.py

```diff
@@ -0,0 +1,168 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from time import time as timestamp
+
+import requests
+
+from datadog_checks.checks import AgentCheck
+from datadog_checks.config import _is_affirmative
+from datadog_checks.utils.containers import hash_mutable
+from .errors import ApiUnreachable
+
+
+class Vault(AgentCheck):
+    CHECK_NAME = 'vault'
+    DEFAULT_API_VERSION = '1'
+    EVENT_LEADER_CHANGE = 'vault.leader_change'
+    SERVICE_CHECK_CONNECT = 'vault.can_connect'
+    SERVICE_CHECK_UNSEALED = 'vault.unsealed'
+    SERVICE_CHECK_INITIALIZED = 'vault.initialized'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        super(Vault, self).__init__(name, init_config, agentConfig, instances)
+        self.api_versions = {
+            '1': {
+                'functions': {
+                    'check_leader': self.check_leader_v1,
+                    'check_health': self.check_health_v1,
+                }
+            },
+        }
+        self.config = {}
+
+    def check(self, instance):
+        config = self.get_config(instance)
+        if config is None:
+            return
+
+        tags = list(config['tags'])
+
+        try:
+            config['api']['check_leader'](config, tags)
```

Comment:
It's a bit weird to have a config that executes a function -- can you add a quick comment here explaining what's going on? I had to bounce around a bit yesterday to figure out what was going on in this line

Reply:
I'll add a comment, thanks! The alternative for handling different API versions would be:
```
if v1:
    func1_v1(...)
    func2_v1(...)
    func3_v1(...)
    ...
elif v2:
    func1_v2(...)
    func2_v2(...)
    func3_v2(...)
    ...
elif v3:
    func1_v3(...)
    func2_v3(...)
    func3_v3(...)
    ...
...
```

Reply:
yea I understand the rationale, it's good it's just also not immedately clear what's going on since usually a confg object doesn't have a function in it :)